src/datasets/arrow_dataset.py

```diff
@@ -5138,6 +5148,34 @@ def push_to_hub(
                 "Failed to push_to_hub: please specify either max_shard_size or num_shards, but not both."
             )
 
+        # Create repo if it doesn't exist
+        api = HfApi(endpoint=config.HF_ENDPOINT)
+        try:
+            api.repo_info(repo_id, token=token)
+        except RepositoryNotFoundError:
+            repo_url = create_repo(hf_api=api, repo_id=repo_id, token=token, exist_ok=True, repo_type="dataset")
+            repo_id = repo_url.repo_id
```

Comment:
This is the correct way of getting the `repo_id` when creating a repository (including potentially resolving the namespace if not explicitly provided) _BUT_ it is only available since `huggingface_hub>=0.12`. Before that, the return value was a pure `str` representing the URL (and has to be manually parsed). I don't know how it was handled in `datasets` before, maybe worth to keep it consistent (e.g. most probably keeping the `repo_id` provided by the user and assume it is the correct one). 

For the record, the new return value is of type `RepoUrl` which provides info like endpoint, repo_type, namespace and repo_id. `RepoUrl` is a subclass of `str` so it doesn't bring any breaking changes.

Reply:
I checked 2.7, 2.6 and 1.18.X for how the output of `create_repo` was parsed, and it does not seem the output is ever used, hence it does not seem there is preferred way to parse the URL? The output is simply `https://huggingface.co/{repo_id}`, so maybe:
``` python
from urllib.parse import urlparse
repo_id = urlparse(repo_url).path[:1]
```

Reply:
Maybe you can move this logic in `_hf_hub_fixes` as well ?

Reply:
Like [this](https://github.com/huggingface/datasets/pull/5528/commits/12327e310b2086a5cb0e51c23677ba6a3102a55a) ?